components/WeatherForecast.js

```diff
@@ -0,0 +1,118 @@
+import {
+  WeatherForecastField,
+  Overlay,
+  StyledDiv,
+  DisplayedDayDetailsPage,
+  ThreeDaysContainer,
+  WeatherIconDetailsPage,
+  DayContainer,
+  ModalContainer,
+  TemperatureDetailsPage,
+  WeatherHeadline,
+} from "./styledComponents/WeatherForecast.styles";
+
+import { useState, useEffect } from "react";
+import useSWR from "swr";
+import WeekdayFromDateString from "./WeekdayFromDateString";
+import WeatherForecastModal from "./WeatherForecastModal";
+
+const API = process.env.NEXT_PUBLIC_WEATHER_API_KEY;
+
+export default function WeatherForecast({ detailActivity }) {
+  const [filteredWeatherMorning, setFilteredWeatherMorning] = useState([]);
+  const [filteredWeatherAfternoon, setFilteredWeatherAfternoon] = useState([]);
+  const [filteredWeatherEvening, setFilteredWeatherEvening] = useState([]);
+  const [showWeatherForecastModal, setShowWeatherForecastModal] =
+    useState(false);
+
+  const { data: weather } = useSWR(
+    detailActivity.lat &&
+      `https://api.openweathermap.org/data/2.5/forecast?lat=${detailActivity.lat}&lon=${detailActivity.lng}&units=metric&appid=${API}`
+  );
+
+  useEffect(() => {
+    if (weather && weather.list) {
+      const filteredWeather = filterWeatherData(weather.list);
+      setFilteredWeatherMorning(
+        filterWeatherByTime(filteredWeather, "06:00:00")
+      );
+      setFilteredWeatherAfternoon(
+        filterWeatherByTime(filteredWeather, "12:00:00")
+      );
+      setFilteredWeatherEvening(
+        filterWeatherByTime(filteredWeather, "18:00:00")
+      );
+    }
+  }, [weather]);
```

Comment:
You don't need this effect. the Component rerenders when the data is received by the useSWR hook. Therefore, any code below the hook is executed again. You can simply call your filter functions directly there:
```js
  const { data: weather, isLoading } = useSWR(
    detailActivity.lat &&
      `https://api.openweathermap.org/data/2.5/forecast?lat=${detailActivity.lat}&lon=${detailActivity.lng}&units=metric&appid=${API}`
  );
  
  if(isLoading) {
    // if weather data is not there yet, stop the function here.
    return "loading..."
  }
  
  const filteredWeather = filterWeatherData(weather.list);
  const filteredWeatherMorning = filterWeatherByTime(filteredWeather, "06:00:00")
  const filteredWeatherAfternoon = filterWeatherByTime(filteredWeather, "12:00:00")
  const filteredWeatherEvening = filterWeatherByTime(filteredWeather, "18:00:00")
```